main.go

```diff
@@ -132,6 +171,60 @@ func main() {
 	}
 }
 
+func deleteAction(ctx context.Context, clientSet client.Client, hr *fluxhelmv2beta1.HelmRelease, timeout, interval time.Duration) {
+	instance := &fluxhelmv2beta1.HelmRelease{}
+	key := client.ObjectKey{
+		Name:      hr.Name,
+		Namespace: hr.Namespace,
+	}
+	if err := clientSet.Get(ctx, key, instance); client.IgnoreNotFound(err) != nil {
+		log.Fatalf("Failed to get instance of the helmrelease with %v", err)
+	} else if err != nil {
+		// This means that the object was not found
+		log.Errorf("Did not find the helm release: %v", hr.Name)
+	} else {
+		instance.Annotations = hr.Annotations
+		instance.Labels = hr.Labels
+		instance.Spec = hr.Spec
+
+		log.Infof("Found the helm release: %v, deleting the release...", hr.Name)
+		// Delete the HelmRelease
+		retry(ctx, func() error { return clientSet.Delete(ctx, instance) }, interval)
+	}
+
+	if err := PollHelmReleaseDeleted(ctx, clientSet, hr, timeout); err != nil {
```

Comment:
This will fail before the timeout you have passed here because the `ctx` will always timeout before the `timeout` that you are passing to `PollHelmReleaseDeleted`. In my opinion, the execution of the entire executor should have a timeout, not just the polling portion. It may be worth looking to see if there is a poller based on a context with a timeout to see if we can keep the same timeout for the entire executor

Reply:
We need some cleanup logic before exiting the executor in that case. Argo workflows does support this - might be worth looking into.
For now we want to carry out the following steps:
1. Delete HelmRelease
2. Wait for HelmRelease to be deleted until timeout (new context maybe?)
3. If timeout, 
  a. Remove finalizer
  b. Wait for HelmRelease and owned objects to be garbage collected
  c. Force delete underlying helm release using `helm uninstall --force` functionality

Reply:
What happens if the delete has to be retried multiple times, I don't quite understand why calling delete and the timeout there should be different from the timeout where we are waiting for the HelmRelease to be deleted by helm-controller